### PR TITLE
clang: add OpenMP library, should now work

### LINKED ISSUE #414: openmp support for clang
Since you've changed the clang version to 3.8, do you have any plan to enable [openmp](http://llvm.org/releases/3.8.0/tools/clang/docs/ReleaseNotes.html#openmp-support-in-clang)?

### PR DESCRIPTION
**Pull request by joakim-noah**
Fixes #414, tested with a C file calling OpenMP, definitely sped up.

### CODE REVIEW

File: packages/libllvm/build.sh [hunk 1]
```diff
@@ -93,6 +96,18 @@ termux_step_post_make_install () {
 	for tool in clang clang++ cc c++ cpp gcc g++ ${TERMUX_HOST_PLATFORM}-{clang,clang++,gcc,g++,cpp}; do
 		ln -f -s clang-${_PKG_MAJOR_VERSION} $tool
 	done
+
+	local OPENMP_ARCH
+	if [ $TERMUX_ARCH = "i686" ] || [ $TERMUX_ARCH = "x86_64" ]; then
+		OPENMP_ARCH="i386"
```
Comment (joakim-noah) [inline packages/libllvm/build.sh]: I guess you're having clang generate x86 code on x64, so just followed suit here.

**Comment by joakim-noah** [comment]
Looks like the CI only fails because libllvm takes forever to build, with Travis killing it before 50 mins.

File: packages/libllvm/build.sh (hunk 1 above)
Comment (fornwall) [inline packages/libllvm/build.sh]: The Termux clang x86-64 produces x86-64 code.

As I understand it one uses `LLVM_TARGET_ARCH=X86` for both x86 and x86-64, but llvm then looks at `LLVM_DEFAULT_TARGET_TRIPLE` to see if 32-bit or 64-bit code should be built.

**Comment by fornwall** [comment]
Nice! A quick check - what is the size increase of the clang&libllvm packages after adding this?

If it's considerable it may be worth splitting up the package, since I guess most clang users does not use OpenMP (but if the size increase is not that big it's probably not worth the trouble).

File: packages/libllvm/build.sh (hunk 1 above)
Comment (joakim-noah) [inline packages/libllvm/build.sh]: OK, I should have looked into that more: you're right, llvm uses `X86` for both i386 and x64 code generation.  I've updated this PR to use the right one.

**Comment by joakim-noah** [comment]
There should be no difference in the libllvm package, as I remove all the static libraries with the first two globs in this patch.  The libomp.a varies from 4-6 MBs in size and is only included in the clang subpackage, so it increases that package by about 800 MB.  Let me know if you want that included with clang or not.

**Comment by joakim-noah** [comment]
@fornwall, just waiting for you to let me know if you prefer OpenMP in its own package or with clang, given the 800 KB increase in the package size.

**Comment by fornwall** [comment]
As you say the compressed (download) size is bumped with 800 KB, and installed size with 6 MB. Let's try to live with this, at least initially :).

The updated version `4.0.1-2` of clang is now available with openmp support - nice work!

**Pull request merged**